R/PIConfiguration.R

```diff
@@ -1,69 +1,73 @@
-        validateIsLogical(value)
+        ospsuite.utils::validateIsLogical(value)
         private$.simulateSteadyState <- value
       }
     },
-    #' @field steadyStateTime Time in minutes to simulate if simulating steady-state. May be NULL
+    #' @field steadyStateTime Time in minutes for steady-state simulation. May be NULL
     steadyStateTime = function(value) {
       if (missing(value)) {
         private$.steadyStateTime
       } else {
-        validateIsNumeric(value)
+        ospsuite.utils::validateIsNumeric(value)
         if (value < 0) {
           stop(paste0("steadyStateTime must be a positive numerical value, but the value is ", value))
         }
         private$.steadyStateTime <- value
       }
     },
-    #' @field printEvaluationFeedback Boolean. If `TRUE`, the objective function value
-    #' will be printed after each function evaluation.
-    #' Default is `FALSE`
+    #' @field printEvaluationFeedback Boolean. If `TRUE`, prints objective
+    #' function value after each evaluation. Default is `FALSE`
     printEvaluationFeedback = function(value) {
       if (missing(value)) {
         private$.printEvaluationFeedback
       } else {
-        validateIsLogical(value)
+        ospsuite.utils::validateIsLogical(value)
         private$.printEvaluationFeedback <- value
       }
     },
 
-    #' @field simulationRunOptions Object of type `SimulationRunOptions` that will be passed
-    #' to simulation runs. If `NULL`, default options are used.
+    #' @field simulationRunOptions Object of `SimulationRunOptions` for simulation
+    #' runs. If `NULL`, default options are used.
     simulationRunOptions = function(value) {
       if (missing(value)) {
         private$.simulationRunOptions
       } else {
-        validateIsOfType(value, "SimulationRunOptions", nullAllowed = TRUE)
+        ospsuite.utils::validateIsOfType(
+          value, "SimulationRunOptions", nullAllowed = TRUE
+        )
         private$.simulationRunOptions <- value
       }
     },
 
-    #' @field targetFunctionType Type of the target function used for error
-    #' calculation. Supported target functions  are listed  in
-    #' `ospsuite.parameteridentification::ObjectiveFunctions`.
-    targetFunctionType = function(value) {
-      if (missing(value)) {
-        private$.targetFunctionType
+    #' @field objectiveFunctionOptions Configuring model fit assessment
+    #' within `.objectiveFunction`. Influences error and model fit calculation.
+    #' Supported options and their allowable values are detailed in
+    #' `ospsuite.parameteridentification::ObjectiveFunctionOptions`.
+    objectiveFunctionOptions = function(options = list()) {
+      if (missing(options)) {
+        private$.objectiveFunctionOptions
       } else {
-        validateIsCharacter(value)
-        validateEnumValue(tolower(value), ObjectiveFunctions)
-        private$.targetFunctionType <- value
+        values <- enumWithValues(ObjectiveFunctionOptions)
```

Comment:
Why is this required? From what I see, `enumWithValues(ObjectiveFunctionOptions)` returns exactly `ObjectiveFunctionOptions`.

Reply:
I realize it was too much and simplifying will make it more clean.